### PR TITLE
Ensure request has valid API key to view sites

### PR DESCRIPTION
**Pull request by RyanMacG**
The current implementation ensures that the request for clients
contains a valid API key before returning the list of IPs. This ensures
we have functional parity in that regard.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -7,13 +7,28 @@ class App < Sinatra::Base
...
       user: ENV.fetch('RR_DB_USER'),
-      password: ENV.fetch('RR_DB_PASS'))
-    tables = db.fetch('show tables')
-    tables.all.to_json
+      password: ENV.fetch('RR_DB_PASS')
+    )
+  end
+
+  def valid_key?
+    params.dig(:key) == ENV.fetch('FRONTEND_API_KEY')
```
Comment (afoldesi-gds) [inline app.rb]: I would add a not nil check on params.dig

File: spec/clients_spec.rb [hunk 2]
```diff
@@ -9,16 +9,42 @@
         password: ENV.fetch('RR_DB_PASS'))
     end
 
-    context 'Given no shared key' do
+    context 'Given no API key' do
```
Comment (afoldesi-gds) [inline spec/clients_spec.rb]: :) Thanks!

File: app.rb (hunk 1 above)
Comment (RyanMacG) [reply app.rb]: calling `.dig(:non_existent)` evaluates to `nil` so it wouldn't be needed

Comment (afoldesi-gds) [reply app.rb]: That's not the point. Mis-configuration of environment variable opens up this to the world if called without a key (nil == nil)

**Pull request merged**